src/main/webapp/app/tasks/task-form/task-form.component.ts

```diff
@@ -1,32 +1,41 @@
 import { ChangeDetectionStrategy, Component, EventEmitter, Inject, Output } from '@angular/core';
 import { FormControl, FormGroup, Validators } from '@angular/forms';
 
+// import moment from 'moment';
 import { Task } from '../task';
 import { TaskService } from '../task.service';
 
 /**
  * A form to create tiny tasks.
  */
+import * as moment from 'moment';
 @Component({
   selector: 'tiny-task-form',
   templateUrl: './task-form.component.html',
-  styleUrls: ['./task-form.component.scss'],
+  styleUrls: ['./task-form.component.scss'], 
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class TaskFormComponent {
 
   @Output() created: EventEmitter<Task> = new EventEmitter();
+  today = new Date();
 
   taskForm: FormGroup = new FormGroup({
-    name: new FormControl('', Validators.required)
+    name: new FormControl('', Validators.required),
+    duedate: new FormControl('', Validators.required)
```

Comment:
From the requirements ("tasks with a due date go to the top of the list, tasks without a due date go to the bottom of the list") the due date should have stayed optional.